metals/src/main/scala/scala/meta/internal/metals/ServerCommands.scala

```diff
@@ -199,6 +216,13 @@ object ServerCommands {
     CascadeCompile,
     CancelCompile,
     BspSwitch,
-    StartDebugAdapter
+    StartDebugAdapter,
+    NewScalaFile
   )
 }
+
+case class MetalsNewScalaFileParams(
+    @Nullable directory: String,
```

Comment:
We should be able to just pack it into Option? Or is it a gson limitation?

Reply:
I just followed other params case class declaration, like `MetalsStatusParams`. I would guess, it's a lsp4j limitation, not sure. Maybe @olafurpg  can comment on this ?
Then I do `Option(directory)`, yes.

Reply:
We're using gson to serialize to JSON and I think you need custom gson adapters to support `Option[T]`

Reply:
The `@Nullable` annotation isn't required, but I used it to document field where I would have preferred to use `Option[T]`